common/app/agents/DeeplyReadAgent.scala

```diff
@@ -1,18 +1,126 @@
 package agents
 
-import common.{Box, GuLogging}
-import model.dotcomrendering.{OnwardCollectionResponse, Trail}
-
-class DeeplyReadAgent extends GuLogging {
-  private val trailsBox = Box[Seq[Trail]](Seq())
-  def onwardsJourneyResponse = {
-    OnwardCollectionResponse(
-      heading = "Deeply read",
-      trails = trailsBox.get(),
+import com.gu.contentapi.client.model.v1.Content
+import com.gu.contentapi.client.utils.CapiModelEnrichment.RenderingFormat
+import common._
+import contentapi.ContentApiClient
+import model.ContentFormat
+import model.dotcomrendering.OnwardCollectionResponse
+import play.api.libs.json._
+import services.{OphanApi, OphanDeeplyReadItem}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import model.dotcomrendering.Trail
+
+import java.lang.Throwable
+
+class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) extends GuLogging {
+
+  /*
+      We use a mutable map instead of a com.gu.Box, as the latter is a minimal wrapper.
+      Both are used as key value store providing in memory caching.
+
+      This implies that several EC2 instances running this app couqld be in slightly different states
+      at any point in time. This is not an issue as we have our CDN caching layer in front.
+   */
+
+  private val deeplyReadItems = Box[Map[Edition, Seq[Trail]]](Map.empty)
+
+  def removeStartingSlash(path: String): String = {
+    if (path.startsWith("/")) path.stripPrefix("/") else path
+  }
+
+  def refresh()(implicit ec: ExecutionContext): Future[Unit] = {
+    log.info(s"Deeply Read Agent refresh()")
+    /*
+      We query Ophan for the deeply read URLs and use them to queryCapi
+      then use this information to create a sequence of trails that we cache
+      using a Box structure.
+     */
+    Future
+      .sequence(Edition.all.map { edition =>
+        ophanApi.getDeeplyRead(edition).flatMap {
+          ophanDeeplyReadItems =>
+            log.info(s"ophanItems updated with: ${ophanDeeplyReadItems.size} new items")
+            val constructedTrail: Seq[Future[Trail]] = ophanDeeplyReadItems.map {
+              ophanItem =>
+                log.info(s"CAPI lookup for Ophan deeply read item: ${ophanItem.toString}")
+                val path = removeStartingSlash(ophanItem.path)
+                log.info(s"CAPI Lookup for path: ${path}")
```

Comment:
Nitpick 🧑‍🏫 IntelliJ tells me the enclosing block is redundant. I think it's only needed when an expression gets evaluated, e.g. in line 45 `${ophanDeeplyReadItems.size}`

```suggestion
                log.info(s"CAPI Lookup for path: $path")
```